inst/OMOP-Summary.Rmd

```diff
@@ -0,0 +1,440 @@
+                                value = nrow(st[["person"]] %>% filter(is.na(year_of_birth)))))
+summary <- bind_rows(summary, c(c(variable = "Number of measurements"),
+                                value = filter(table_counts, table == "measurement") %>% pull(n)))
+summary <- bind_rows(summary, c(c(variable = "Number of observations"), 
+                                value = filter(table_counts, table == "observation") %>% pull(n)))
+summary <- bind_rows(summary, c(c(variable = "Number of conditions"), 
+                                value = filter(table_counts, table == "condition_occurrence") %>% pull(n)))
+summary <- bind_rows(summary, c(c(variable = "Number of procedures"), 
+                                value = filter(table_counts, table == "procedure_occurrence") %>% pull(n)))
+summary <- bind_rows(summary, c(c(variable = "Number of drug exposures"), 
+                                value = filter(table_counts, table == "drug_exposure") %>% pull(n)))
+```
+
+## General counts
+```{r}
+summary %>%
+  mutate(value = as.integer(value)) %>% 
+  print_large_kable()
+```
+
+<br>
+<br>
+
+## Person Table
+
+```{r ages}
+if ("person" %in% zero_vals) {
+  cat("There is no data in the person table")
+} else {
+  age_plot <- plot_age(overview)%>% custom_theme(params$custom_colors)
+  check_age <- min(overview$age, na.rm = TRUE) >= 18
+
+  sex_plot <- plot_sex(st[["person"]]) %>% custom_theme(params$custom_colors)
+  ethnic_plot <- plot_ethnicity(st[["person"]])%>% custom_theme(params$custom_colors)
+
+  age_plot + sex_plot + ethnic_plot
+}
+```
+
+<br>
+<br>
+
+## Visit Occurence
+
+Number of visits stratified by visit type:
+
+```{r visit-types}
+if (!("visit_occurrence" %in% zero_vals)) {
+  st[["visit_occurrence"]] %>%
+    group_by(visit_concept_id) %>%
+    summarise(
+      count = n(),
+      percentage = round((n()/nrow(st[["visit_occurrence"]]))*100, 0)
+    ) %>% 
+    print_large_kable()
+}
+```
+
+```{r visit-profile}
+plot_visit_profile(st[["visit_occurrence"]],params$start_date,params$end_date,params$custom_colors)%>% custom_theme(params$custom_colors)
+```
+
+```{r admission-from1}
+st[["visit_occurrence"]] %>%
+  group_by(admitted_from_concept_id) %>%
+  tally() %>%
+  print_large_kable()
+```
+
+```{r admission-from2}
+admission_from_plot <- st[["visit_occurrence"]] %>%
+  mutate(admitted_from_concept_id = forcats::fct_infreq(as.character(admitted_from_concept_id))) %>%
+  group_by(admitted_from_concept_id) %>%
+  tally() %>%
+  check_zero_tally(n) %>%
+  ggplot(aes(x = admitted_from_concept_id)) +
+  geom_point(aes(y = n, colour = .is_zero)) +
+  geom_segment(aes(
+    y = 0,
+    yend = n,
+    xend = admitted_from_concept_id)) +
+  scale_colour_manual(values = params$custom_colors, drop = FALSE) +
+  theme_classic() +
+  labs(y = "Count", x = "Type") +
+  theme(
+    legend.position = "none",
+    plot.title.position = "plot",
+    axis.title.y = element_blank()) +
+    coord_flip() +
```

Comment:
You should be able to dump most of this into a theme_function.
There's one here already called decovid theme (I flagged it in a comment above).
And there's another example of how it's done in the reports package (it's called custom_theme).

Reply:
This is done!